### ISSUE TITLE
[Feature Request] make where varargs to specify logic precedence

### ISSUE THREAD

**Issue by tsouza** [snd/mohair#19]
Hi,

The following query:

```
user.select("*").
    where({ username: "user", password: "hash",
        $or: [ { active: 1 }, { active: { $null: true } } ]});
```

Would produce the following SQL:

```
 SELECT * FROM user WHERE username = ? AND password = ? AND (active IS NULL) OR (active = ?)
```

In my case, this is not the intended SQL as "(active IS NULL) OR (active = ?)" needs precedence (i.e. between parentheses), so I suggest that where method to be varags:

```
user.select("*").
    where({ username: "user", password: "hash"}, 
        { $or: [ { active: 1 }, { active: { $null: true } } ]});
```

And this would produce the following:

```
 SELECT * FROM user WHERE (username = ? AND password = ?) AND ((active IS NULL) OR (active = ?))
```

Att

**Comment by snd** [comment]
your initial query was correct. there should be another pair of parentheses around the $or.
i am going to fix this soon.

**Comment by tsouza** [comment]
Hi there,

```
 Any ETA on fixing this?
```

Regards

**Comment by snd** [comment]
fixed in version 0.9.2.

i am sorry to have left you waiting so long.

**Comment by tsouza** [comment]
Hi,

```
No problem!!! Thanks for you support =D
```

Cheers
